zhmcclient/_partition_link.py

```diff
@@ -0,0 +1,786 @@
+
+        Parameters:
+
+          name (:term:`string`): Filter pattern (regular expression)
+            to limit returned partitions to those that have a matching
+            name. If `None`, no filtering for the partition name takes place.
+
+          status (:term:`string`): Filter string to limit returned
+            partitions to those that have a matching status. The value
+            must be a valid partition 'status' property value. If `None`, no
+            filtering for the partition status takes place.
+
+        Returns:
+
+          List of :class:`~zhmcclient.Partition` objects representing the
+          partitions to which this partition link is currently attached, with
+          a minimal set of properties ('object-id', 'name').
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        part_list = []
+        part_uri_list = []
+        pl_type = self.get_property('type')
+        if pl_type == 'ctc':
+            for path in self.get_property('paths'):
+                for device in path.get('devices'):
+                    for endpoint in device.get('endpoint-pair'):
+                        part_uri = endpoint.get('partition-uri')
+                        part_name = endpoint.get('partition-name')
+                        part = self.cpc.partitions.resource_object(
+                            part_uri, {'name': part_name})
+                        if name:
+                            if not re.match(name, part_name):
+                                continue
+                        if status:
+                            if part.get_property('status') != status:
+                                continue
+                        if part_uri not in part_uri_list:
+                            part_uri_list.append(part_uri)
+                            part_list.append(part)
+        else:
+            # Hipersockets or SMC-D
+            for bc in self.get_property('bus-connections'):
+                part_uri = bc.get('partition-uri')
+                part_name = bc.get('partition-name')
+                part = self.cpc.partitions.resource_object(
+                    part_uri, {'name': part_name, 'nics': []})
+                if name:
+                    if not re.match(name, part_name):
+                        continue
+                if status:
+                    if part.get_property('status') != status:
+                        continue
+                if part_uri not in part_uri_list:
+                    part_uri_list.append(part_uri)
+                    part_list.append(part)
+        return part_list
+
+    def attach_to_partition(
+            self, partition, properties, wait_for_completion=True,
+            operation_timeout=None):
+        """
+        Attach this partition link to a partition, creating corresponding
```

Comment:
We have the attach tape link/storage group operations on the Partition resource. I am wondering whether it makes sense to move the attach partition link operation there as well for consistency.